client/graphics/lights.hpp

```diff
@@ -13,23 +13,17 @@
 
 namespace tec {
 	struct BaseLight {
-		glm::vec3 color;
-		float ambient_intensity;
-		float diffuse_intensity;
-
-		BaseLight() {
-			color = glm::vec3(1.0f, 1.0f, 1.0f);
-			ambient_intensity = 0.5f;
-			diffuse_intensity = 0.2f;
-		}
+		glm::vec3 color{ 1.f, 1.f, 1.f };
+		float ambient_intensity{ 0.5f };
+		float diffuse_intensity{ 0.2f };
+
+		BaseLight() = default;
 	};
 
 	struct DirectionalLight : public BaseLight {
-		glm::vec3 direction;
+		glm::vec3 direction{ 0.f, 0.f, -1.f };
```

Comment:
In this case I'm okay with using brace initialisation, but I'd still prefer `glm::vec3 direction = {0.0f, 0.0f, -1.0f};`

I'm also not really a fan of leaving out the `0` there just because we can. I wrote the above as `0.1f, 0.0f, -1.0f` at first and had to look at this line three times before I realised it's just `0`.

Reply:
For future I will include both sides.